### PR TITLE
shiftstack: replace centos8 by centos9

### PR DESCRIPTION
**Pull request by EmilienM**
We use CentOS for bastion proxy & also CI liveliness checks, let's use
the latest CentOS 9 stream version.

### CODE REVIEW

**Comment by EmilienM** [comment]
There is definitely something wrong:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/26766/rehearse-26766-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.9-e2e-openstack-proxy/1499941165435195392/artifacts/e2e-openstack-proxy/openstack-provision-bastionproxy/build-log.txt

File: ci-operator/step-registry/openstack/deprovision/bastionproxy/openstack-deprovision-bastionproxy-commands.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ if [[ -f ${SHARED_DIR}"/BASTION_FIP" ]]; then
   # configure the local container environment to have the correct SSH configuration
   if ! whoami &> /dev/null; then
       if [[ -w /etc/passwd ]]; then
-          echo "${BASTION_USER}:x:$(id -u):0:${BASTION_USER:-centos} user:${HOME}:/sbin/nologin" >> /etc/passwd
+          echo "${BASTION_USER-cloud-user}:x:$(id -u):0:${BASTION_USER:-cloud-user} user:${HOME}:/sbin/nologin" >> /etc/passwd
```
Comment (pierreprinetti) [inline ci-operator/step-registry/openstack/deprovision/bastionproxy/openstack-deprovision-bastionproxy-commands.sh:32]: Do you want to use :- here for consistency?

Comment (mandre) [reply ci-operator/step-registry/openstack/deprovision/bastionproxy/openstack-deprovision-bastionproxy-commands.sh:32]: There's a typo, should be
```suggestion
          echo "${BASTION_USER:-cloud-user}:x:$(id -u):0:${BASTION_USER:-cloud-user} user:${HOME}:/sbin/nologin" >> /etc/passwd
```

Comment (EmilienM) [reply ci-operator/step-registry/openstack/deprovision/bastionproxy/openstack-deprovision-bastionproxy-commands.sh:32]: @mandre https://github.com/openshift/release/pull/27339

**Review by mandre — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**